hooks/queries/tokenMetadata.ts

```diff
@@ -0,0 +1,40 @@
+import { findMetadataPda } from '@metaplex-foundation/js'
+import { Metadata } from '@metaplex-foundation/mpl-token-metadata'
+import { PublicKey } from '@solana/web3.js'
+import { useQuery } from '@tanstack/react-query'
+import useWalletStore from 'stores/useWalletStore'
+
+export const useTokenMetadata = (
+  mint: PublicKey | undefined,
+  enableConditions = true
+) => {
+  const connection = useWalletStore((s) => s.connection)
+
+  const enabled = !!mint && !!enableConditions
+
+  const query = useQuery({
+    queryFn: async () => {
+      const mintPubkey = new PublicKey(mint!)
+      const metadataAccount = findMetadataPda(mintPubkey)
+      const accountData = await connection.current.getAccountInfo(
+        metadataAccount
+      )
+
+      const state = Metadata.deserialize(accountData!.data)
+      const jsonUri = state[0].data.uri.slice(
+        0,
+        state[0].data.uri.indexOf('\x00')
+      )
+
+      const data = await (await fetch(jsonUri)).json()
+      //we dont want unsafe imgs
```

Comment:
i dont follow this comment -- are imgs being filtered here?

Reply:
img is inside data. will adjust comment for people to not include imgs from metadata

Reply:
i think it would be better if this query just returned the data in its natural state (`Metadata`) and then the components themselves were responsible for using the right source of data for the images. you could leave a comment tho. i think by default any transformation of data is bad so better to keep to a minimum. thats how i'd do it anyway.